src/LEGO.AsyncAPI/Writers/SpecialCharacterStringExtensions.cs

```diff
@@ -183,11 +191,25 @@ internal static string GetYamlCompatibleString(this string input)
                 return $"'{input}'";
             }
 
-            // If string can be mistaken as a number, a boolean, or a timestamp,
-            // wrap it in quote to indicate that this is indeed a string, not a number, a boolean, or a timestamp
-            if (decimal.TryParse(input, NumberStyles.Float, CultureInfo.InvariantCulture, out var _) ||
-                bool.TryParse(input, out var _) ||
-                DateTime.TryParse(input, out var _))
+            // Handle lexemes that can be intperated as as string
+            // https://yaml.org/spec/1.2-old/spec.html#id2761292 
+            switch (input.ToLower())
+            {
+                // Example 2.20. Floating Point
+                case "-.inf":
+                case ".inf":
+                case ".nan":
```

Comment:
These cases were all missed

Reply:
good find!
I think the datetime case has gone missing however.

Reply:
might get caught by the forbidden combinations though. - dont remember off the top of my head.

Reply:
It should be covered by the regex however just because I am not 100% sure let me add a test to explicitly check for that.